application/android/app/src/main/java/com/group12/beabee/network/ServiceAPI.java

```diff
@@ -127,4 +130,42 @@
 
     @POST("entities/question")
     Call<BasicResponse> createQuestion(@Body Question question);
+
+
+
+    //GROUP GOAL LINKS:
+    @PUT("groupgoals")
+    Call<BasicResponse> updateGG(@Body GroupGoalDTO userDTO);
+
+    @GET("groupgoals/{groupgoal_id}")
+    Call<GroupGoalDTO> getGGDetail(@Path("groupgoal_id") int goalId);
+
+    @DELETE("groupgoals/{groupgoal_id}")
+    Call<BasicResponse> deleteGG(@Path("groupgoal_id") int goalId);
+
+    @PUT("groupgoals/{groupgoal_id}/members")
+    Call<BasicResponse> addMember(@Body GroupGoalDTO goalDTO, UserDTO userDTO);
+
+    @POST("groupgoals/{groupgoal_id}/token")
+    Call<BasicResponse> createToken(@Path("groupgoal_id") int goalId);
+
+    @POST("groupgoals/{user_id}")
+    Call<BasicResponse> createGG(@Path("user_id") int user_id, @Body GroupGoalDTO ggDTO);
+
+    @DELETE("groupgoals/{user_id}/{groupgoal_id}")
+    Call<BasicResponse> leaveGG(@Path("user_id") int userId, @Path("groupgoal_id") int goalId);
+
+    @POST("groupgoals/{user_id}/join")
+    Call<BasicResponse> joinGG(@Path("user_id") int user_id,@Query("token") String token);
+
+    @GET("groupgoals/created_by/{user_id}")
+    Call<List<GroupGoalDTO>> getAllGGCreatedByUser(@Path("user_id") int user_id, @Body GroupGoalDTO ggDTO);
```

Comment:
Here on the respond type instead of `GroupGoalDTO` the server sends an object with fewer fields, on the other requests we used models which are called `___Short` to capture the necessary information, you can also create a short version of the `GroupGoalDTO` which can be called `GroupGoalShort` and use it instead